### ISSUE TITLE
NZB Club alternative

### ISSUE THREAD

**Issue by poo706** [8cylinder/tv-overlord#50]
It seems as though NZB Club didn't index last night's shows, but NZB Index did.  I see that there's a nzbindex_com.py, but it's unclear to me how to get tvol to use it.

**Comment by 8cylinder** [comment]
nzbindex's rss feed stopped working a while ago, it throws a 503 error.  I suspect its because the owners of the site have stopped putting any work into maintaining it.  I kept the nzbindex_com.py around in case the site started working again.

I could change the code to scrape the search results page (which I do on some torrent sites) instead of using an rss feed, but I prefer not to do that because if they ever change their design the scraping code stops working.  If you know of any good sites where you can do a search and save it as an rss link I'll add it to tv overlord.

That being said, if nzbindex is one of the better ones, I'll get it working.  There is only one nzb indexer being used right now and there should be more.

**Comment by poo706** [comment]
I really don't know if nzbindex is any better.  Other than that one single time 2 weeks ago, I've never had an issue with nzbclub since using tvol.  What about binsearch.info?

**Comment by vergessen** [comment]
Is there any plan to support a generic newznab api?  opens up a lot more nzb sites

**Comment by 8cylinder** [comment]
That sounds interesting, tell me more!  How would that work?  Can you point me to some documentation?

**Comment by vergessen** [comment]
There are several sites that run newznab as their base software. they do the indexing and provide an api interface to their users http://newznab.readthedocs.io/en/latest/misc/api/ is the docs
https://drunkenslug.com/ and http://fastnzb.info/ are examples but there are also a lot of member type sites that utilize this interface.

**Comment by poo706** [comment]
I saw the commits for "Add support for Newznab and nZEDb".  These aren't reflected in the latest pypi version, right?  Are they functional yet?  I git cloned the repo the other day and managed to run tvol from it, but I couldn't figure out how to get tvol to use nzb.py instead of nzbclub.  Also I couldn't figure out how to add the site, apikey, etc that it looks like nzb.py needs.

**Comment by 8cylinder** [comment]
Yeah, sorry about that, I should have posted this sooner.  Here's the info on it:

## Newznab & nZEDb support

There is support for Newznab and nZEDb servers now.  Its beta so I haven't sent out an update message, but its available in pypi.  I'm going to post the documentation for using it here for now, and not in the wiki.  Once some people have tried it out for a while, I'll make and official release an update the wiki docs.

Since all of these sites require you to be a member to use, I've left the nzbclub search engine active.  It doesn't require membership to use but it doesn't have the best search results.  Also, right now the site is not working.

**How to use it:**

First you have to get an account with a site that uses Newznab or nZEDb.

Update to the latest tvol version (1.4.9) using pip.

Add a section in your config.ini for each site you are signed up for.  There can be more that one.

```
[nzb:drunken]
shortname: DS
longname: DrunkenSlug
url: https://drunkenslug.com
apikey: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

[nzb:nzbplanet]
shortname: NP
longname: NZBPlanet
url: https://api.nzbplanet.net
apikey: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

Each section has to start with `nzb:` and then a name. It doesn't matter what that name is as long as it's different from any others.  It can have spaces in it as well.

Then four fields for each section.

1. `shortname`.  This is three characters or less that will show beside each search result in the table so you know where that search is from.
2. `longname`.  This name will be used if there is an error from the server.  There a lot of different messages that might show, your api limit has been reached, the url is wrong, etc.
3. `url`.  This is the api url to use.
4. `apikey`.

The api url can be different from the url for the site.  For a site that has a url like: `www.somesite.com`, the api url might be `api.somesite.com`.

The apikey is a ~32 character long string and you should be able to get it from your user account details page.

**Comment by poo706** [comment]
I got NZB.ag, which is nZEDb, going without any issues.  Great instructions and nice work as always!  I tried adding "NZBClub" to the "blacklist: " line, but it doesn't seem to work for usenet sites like it does BT sites.

**Comment by 8cylinder** [comment]
Right.  A fix for that will be in the next commit.

**Comment by 8cylinder** [comment]
One issue remaining is how to sort the results.  If someone uses more that one site, the results are stacked as they come in.  This means that the end of the list might not be displayed if there are more that the results table is set to show.

I haven't tried to solve this since I'm don't know if someone would use more that one site.  Something to make note of.  This may or may not be an issue for people.

**Comment by vergessen** [comment]
People do use multiple sites. It is standard practice to have some over lap
in case one indexer goes down or perhaps indexes different content.

**Comment by poo706** [comment]
I updated to the latest .10 version and have "blacklist: NZBClub" in my config.ini, but it still seems to be searching it.  Is that not right?

**Comment by 8cylinder** [comment]
Right, it shouldn't be searching fo nzbclub.  A couple of things to try:

1. You have `blacklist: nzbclub` in the `[App Settings]` section.
2. Have you added that to a different config set than the one you are using? 
3. Try using `tvol download --exclude nzbclub`

The config.ini and on the commandline works for me.  

If none of these work for you, paste the output of `tvol config` (delete any ip address if there are any).
Also tell me the python version and your os your using.

**Comment by poo706** [comment]
Moving the blacklist line up higher in the config.ini seemed to do the trick.  Thanks!
